homeassistant/components/prosegur/__init__.py

```diff
@@ -0,0 +1,58 @@
+"""The Prosegur Alarm integration."""
+import asyncio
+import logging
+
+from pyprosegur.auth import Auth
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client
+
+from .const import CONF_COUNTRY, DOMAIN
+
+PLATFORMS = ["alarm_control_panel"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Prosegur Alarm from a config entry."""
+    try:
+        session = aiohttp_client.async_get_clientsession(hass)
+        hass.data.setdefault(DOMAIN, {})
+        hass.data[DOMAIN][entry.entry_id] = Auth(
+            session,
+            entry.data[CONF_USERNAME],
+            entry.data[CONF_PASSWORD],
+            entry.data[CONF_COUNTRY],
+        )
+        await hass.data[DOMAIN][entry.entry_id].login()
+
+        for component in PLATFORMS:
+            hass.async_create_task(
+                hass.config_entries.async_forward_entry_setup(entry, component)
+            )
+
+        return True
```

Comment:
This should be placed outside of the try-except scope.
If the exceptions below throw, the return False won't be occuring.